mls-rs/src/group/mod.rs

```diff
@@ -1801,8 +1886,8 @@ where
         #[cfg(feature = "by_ref_proposal")]
         self.state.proposals.clear();
 
-        // Clear the pending updates list
-        #[cfg(feature = "by_ref_proposal")]
+        // Clear the pending updates list (but only if they can't show up in a Replace)
+        #[cfg(all(feature = "by_ref_proposal", not(feature = "replace_proposal")))]
         {
             self.pending_updates = Default::default();
         }
```

Comment:
Actually:
1. LeafNodes in Commits never end up in this cache
2. We should delete the state entry for a LeafNode that is committed via Update or Replace is deleted when that proposal is applied.  I had thought this PR added that, but it doesn't seem to.  I'll make the change to do this.

With (2), you actually get the same GC effect as here as long as everything gets committed.

Reply:
On trying to implement, it seems that deleting on epoch change is the right answer.  I have updated that code so that if Replace is allowed, we delete:

1. Any state corresponding to the current leaf node (which has already been used)
2. Any state that was added in an epoch earlier than the current leaf node (which will be stale)